### PR TITLE
Feature/sqzhang/ldas coup

### PR DESCRIPTION
**Pull request by JustinPerket**
I believe this is the branch @saraqzhang referred to in #84. Will test according to GEOSgcm_GC guidelines

### CODE REVIEW

**Comment by JustinPerket** [comment]
@sdrabenh @lltakacs , 
@saraqzhang's Land increment code has passed 0 diff tests for LDAS and GCM. 

My tests of this branch for AMIP, Regular Replay, 0-Increment Replay, and Regression (for Old and New Land) are at:
```
/discover/nobackup/projects/gmao/g6dev/jperket/tests/JPTest_Test_NL_2019-10-08
/discover/nobackup/projects/gmao/g6dev/jperket/tests/JPTest_Test_OL_2019-10-08
```

(Controls I used are:
```
/discover/nobackup/projects/gmao/g6dev/jperket/tests/JPTest_Develop_NL_2019-10-08
/discover/nobackup/projects/gmao/g6dev/jperket/tests/JPTest_Develop_OL_2019-10-08
```
)

**Comment by sdrabenh** [comment]
@wmputman please let me know your input. I verified zero-diff results for restarts and standard history collections.

**Comment by wmputman** [comment]
With the transition now to GIT it will require that we formalize some processes for getting large code updates onto develop.  With large code changes like this we need to have a code summary and review with appropriate attendance from the relevant gatekeepers.  For this request we need at leas t @wmputman @JustinPerket @saraqzhang @sdrabenh and @lltakacs at the meeting.  I'd like to target Monday afternoons for these code meetings. I will send a meeting request for 12:30pm Oct 21.

**Comment by JustinPerket** [comment]
@gmao-rreichle may also want to be at this meeting

**Comment by mathomp4** [comment]
@saraqzhang @JustinPerket The CircleCI failed for a useful reason! Huzzah! It failed because of this code:

https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/94221062a693f6d1bb0c566702fc3bf5f5660290/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/GEOS_CatchGridComp.F90#L2821-L2828

What you should do is change this to:

```fortran
IAU_NAMES = [character(len=11) :: "TCFSAT_IAU", "TCFTRN_IAU", "TCFWLT_IAU", &
                     "QCFSAT_IAU", "QCFTRN_IAU", "QCFWLT_IAU", &
...
```
where the `len=11` is the length of the longest string in the constructor. (I think it's 11, but I might have counted wrong.)

The reason this failed is that gfortran assumes all members of an array constructor are as long as its first member or shortest member. Something like that. So the first is 10 characters long, but eventually it sees a 9-character  (or 11-character) and fails.

I think @JustinPerket can fix this quickly in browser. Also, if you do, please use the `[ ]` version of constructor. It's just nicer. :smile:

**Comment by mathomp4** [comment]
Passed! Thanks @JustinPerket

**Comment by saraqzhang** [comment]
good for me to learn about CircleCI failed reason. thanks!

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/GEOS_CatchGridComp.F90 [hunk 1]
```diff
@@ -4929,6 +5213,269 @@ subroutine Driver ( RC )
...
+                   GHTCNT_IAU(6,:) = GHTCNT6_IAU
+
+                   WESNN_IAU (1,:) = WESNN1_IAU
+                   WESNN_IAU (2,:) = WESNN2_IAU
+                   WESNN_IAU (3,:) = WESNN3_IAU
+
+                   HTSNNN_IAU(1,:) = HTSNNN1_IAU
+                   HTSNNN_IAU(2,:) = HTSNNN2_IAU
+                   HTSNNN_IAU(3,:) = HTSNNN3_IAU
+                   call WRITE_PARALLEL('LDAS_coupling: Calling apply_catch_iau ')
```
Comment (JustinPerket) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/GEOS_CatchGridComp.F90]:
```suggestion
                   SNDZN_IAU (1,:) = SNDZN1_IAU
                   SNDZN_IAU (2,:) = SNDZN2_IAU
                   SNDZN_IAU (3,:) = SNDZN3_IAU
                   
                   call WRITE_PARALLEL('LDAS_coupling: Calling apply_catch_iau ')
```

**Comment by JustinPerket** [comment]
@sdrabenh, I would say the commits after your tests  ( dd89651 to 7c1b607d5 ) are trivially 0-diff. And they are wrapped in the If statement controlled by @saraqzhang's `LDAS_IAU` flag, which is off by default anyways. Is there any further testing to do, assuming the CircleCI tests pass?

**Pull request merged**